Rules/UseCompatibleTypes.cs

```diff
@@ -0,0 +1,854 @@
+            if (unresolved)
+            {
+                diagnosticRecords.Add(new DiagnosticRecord(
+                    String.Format(
+                        Strings.UseCompatibleTypesUnresolvedError,
+                        fullTypeName),
+                    extent,
+                    GetName(),
+                    GetDiagnosticSeverity(),
+                    scriptPath,
+                    null,
+                    null));
+            }
+            else
+            {
+                // Here we are just including the type accelerator name so the type will be easier to spot in the script
+                // on the line number we provide in the diagnostic record.
+                string accelerator = "";
+                if (typeAccelerator)
+                {
+                    try
+                    {
+                        accelerator = " (" + astObject.TypeName?.ToString() + ")";
+                    }
+                    catch { }
+                }
+
+                // Create diagnostic record.
+                diagnosticRecords.Add(new DiagnosticRecord(
+                    String.Format(
+                        Strings.UseCompatibleTypesError,
+                        fullTypeName + accelerator,
+                        platformInfo.PSEdition,
+                        platformInfo.PSVersion,
+                        platformInfo.OS),
+                    extent,
+                    GetName(),
+                    GetDiagnosticSeverity(),
+                    scriptPath,
+                    null,
+                    null));
+            }
+        }
+
+        /// <summary>
+        /// Process errors and add to diagnosticRecords list.
+        /// </summary>
+        private void ProcessErrors()
+        {
+            foreach (ErrorObject error in errorList)
+            {
+                string message  = null;
+
+                if (error.errorRecords != null)
+                {
+                    foreach (ParseError pe in error.errorRecords)
+                    {
+                        message += pe.Message + " ";
+                    }
+                }
+                else
+                {
+                    message = error.message;
+                }
+                
+                diagnosticRecords.Add(new DiagnosticRecord(
+                    String.Format(
+                        Strings.UseCompatibleTypesException,
+                        error.extent.ToString(),
+                        error.errorType,
+                        message),
+                    error.extent,
+                    GetName(),
+                    GetDiagnosticSeverity(),
+                    scriptPath,
+                    null,
+                    null));
+            }
+        }
+        /// <summary>
+        /// Initialize data structures needed to check type compatibility.
+        /// </summary>
+        private void Initialize()
+        {
```

Comment:
Perhaps the initialisation check should be performed here, so the caller can call `Initialize()` without needing to do any checking